### PR TITLE
[Authz] Extract catalog for DataSourceRelationV2 in buildQuery and  pass catalog of queries and commands to PrivilegeObject

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

subtask of #3586 .

1. passing catalog field from `Table` and `Database` to PrivilegeObject for privilege checks. As for`Function` related commands,  no catalog supported other than spark_catalog or v1 catalog in Spark 3.1-3.3.
2. extract catalog from `DataSourceRelationV2` in buildQuery with added `CatalogPluginOptionCatalogExtractor`
3. add ut for v2commands SetNamespaceProperties/CreateNamespace/SetNamespaceLocation/DescribeNamespace/DropNameSpace to verify related PrivilegeObject extraction with catalog in spark 3.3
4. in future PR,  SparkRangerAdminPlugin will use catalog-related instances for privilege checks using catalog info from PrivilegeObject 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @yaooqinn

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala [hunk 1]
```diff
@@ -155,15 +161,19 @@ abstract class PrivilegesBuilderSuite extends AnyFunSuite
   }
 
   test("CreateDatabaseCommand") {
+    assume(isSparkVersionAtMost("3.2"))
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala:164]: what's wrong with these test after such a change?

Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala:164]: We add catalog assertion in thi ut.

In spark 3.0-3.2,  `CREATE DATABASE` sql is parsed into `CreateDatabaseCommand`, which is v1 command and contains no catalog info . And this ut asserts `catalog` in  parsed PviilegeObject is None.

 In spark 3.3, the same ddl will is pared into `CreateNamespace` of v2commands instead, and the asserted catalog is extracted by another related ResolvedDBObjectNameDatabaseExtractor with real value from CatalogPlugin. New ut for this case is also added to `V2CommandsPrivilegesSuite.scala` in this PR.

Comment (yaooqinn) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala:164]: in this case, we shall not put a uncommon test case in the base trait.

Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala:164]: I think they are actually testing different command plans and they are good enough to reflect spark version variation impact for this ddl.

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala (hunk 1 above)
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala:164]: Plus, the migration from v1 to v2 is generally refactoring command detail itself affecting same commands to all and not related to v2 plans.

Comment (yaooqinn) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilderSuite.scala:164]: thanks for the explanation. I want to eliminate spark version and v1/v2 table testing, maybe it's too early to do this.

**Pull request closed without merging**